assume/common/base.py

```diff
@@ -176,14 +177,20 @@ def calculate_generation_cost(
             start (datetime.datetime): The start time for the calculation.
             end (datetime.datetime): The end time for the calculation.
             product_type (str): The type of product for which the generation cost is to be calculated.
-
         """
-        if start not in self.index:
-            start = self.index[0]
+        if end != self.index[-1]:
+            return  # do nothing before the end of the simulation
+
+        start = self.index[0]
```

Comment:
Noone will know, that this basically makes sure that it only runs at the end and therefore runst from the start on..

1. add a param "run_at_end_only" and put line 181-184 into a `if run_at_end_only` check - this makes things more clear
2. refactor the `calculate_generation_cost` function to be only run in `on_stop` of the UnitsOperator for the whole time once..

Either way is fine for me

Reply:
I have having the problem that using this approach only the last value is written into db. I tried the on_stop idea, but the output role is killed before this one, so doesn't work as well. Do you have a suggestion?